website/docs/core-concepts/vendoring/vendoring.md

```diff
@@ -0,0 +1,251 @@
+---
+title: Vendoring
+description: Use Atmos vendoring to make copies of 3rd-party components, stacks, and other artifacts in your own repo.
+sidebar_position: 14
+sidebar_label: Vendoring
+id: vendoring
+---
+
+Atmos natively supports the concept of "vendoring", which is making copies of the 3rd party components, stacks, and other artifacts in your own repo.
+
+The vendoring configuration is described in the `vendor.yaml` manifest, which should be placed in the directory from which the `atmos vendor pull`
+command is executed, usually in the root of the infrastructure repo.
+
+After defining the `vendor.yaml` manifest, all the remote artifacts can be downloaded by running the following command:
+
+```bash
+atmos vendor pull
+```
+
+To vendor a particular component or other artifact, execute the following command:
+
+```bash
+atmos vendor pull -c <component>
+```
+
+<br/>
+
+:::tip
+Refer to [`atmos vendor pull`](/cli/commands/vendor/pull) CLI command for more details
+:::
+
+## Vendoring Manifest
+
+To vendor remote artifacts, create a `vendor.yaml` file similar to the example below:
+
+```yaml
+apiVersion: atmos/v1
+kind: AtmosVendorConfig
+metadata:
+  name: example-vendor-config
+  description: Atmos vendoring manifest
+spec:
+  # `imports` or `sources` (or both) must be defined in a vendoring manifest
+  imports:
+    - "vendor/vendor2.yaml"
+    - "vendor/vendor3.yaml"
+
+  sources:
+    # `source` supports the following protocols: OCI (https://opencontainers.org), Git, Mercurial, HTTP, HTTPS, Amazon S3, Google GCP,
+    # and all URL and archive formats as described in https://github.com/hashicorp/go-getter.
+    # In 'source', Golang templates are supported  https://pkg.go.dev/text/template.
+    # If 'version' is provided, '{{.Version}}' will be replaced with the 'version' value before pulling the files from 'source'.
+    # Download the component from the AWS public ECR registry (https://docs.aws.amazon.com/AmazonECR/latest/public/public-registries.html).
+    - component: "vpc"
```

Comment:
Is there anyway to vendor a component and copy a local file over a vendored file? Such as if I'm using my own `providers.tf` mixin and I want to continue to vendor components without manually overwriting the mixin using `cp` ?

Reply:
Not in a DRY way. The best you can do is `exclude` the `providers.tf` so it won't overwrite your local one.

Reply:
nvm, this is already implemented using `mixins`. thank you.

Reply:
Maybe we could consider a `post_pull_hook` or something like that can run a command. Then you can do the `cp` or anything else. Dont' think we want to support a specialized case around vendoring and then copying another file.

Reply:
You can have another `source` and `target` defined to vendor it from a local file, but that will be ugly (non-DRY)

Reply:
You mean something like this

```yaml
    - component: "vpc-flow-logs-bucket"
      source: "github.com/cloudposse/terraform-aws-components.git//modules/vpc-flow-logs-bucket?ref={{.Version}}"
      version: "1.323.0"
      targets:
        - "components/terraform/infra/vpc-flow-logs-bucket/{{.Version}}"
    - component: "vpc-flow-logs-bucket-mixins"
      # if in components/terraform/infra/vpc-flow-logs-bucket/{{.Version}}/
      source: "../../../../../mixins/providers.tf"
      version: "1.323.0"
      targets:
        - "components/terraform/infra/vpc-flow-logs-bucket/{{.Version}}"
```

It would be nice if `source` then was also a list, no?

```yaml
    - component: "vpc-flow-logs-bucket"
      source:
        - "github.com/cloudposse/terraform-aws-components.git//modules/vpc-flow-logs-bucket?ref={{.Version}}"
        - "../../../../../mixins/providers.tf"
      version: "1.323.0"
      targets:
        - "components/terraform/infra/vpc-flow-logs-bucket/{{.Version}}"
```